### DISCUSSION TITLE
How to build rounded-corners frameless windows?

### DISCUSSION THREAD

**Discussion by xuchaoqian** [tauri-apps/tauri#3219] [Q&A] [answered]
If we disable decorations,  the windows will like this:

![image](https://user-images.githubusercontent.com/1064052/149608265-2157c432-71cc-4068-beb1-0c22fa0121a5.png)

So how how to build rounded-corners frameless windows?

**Comment by xuchaoqian** [comment, marked as answer]
Maybe this is the only way to get this:
![image](https://user-images.githubusercontent.com/1064052/149609926-d34eeb27-c973-4659-bd79-e149c979c656.png)

**Reply by amrbashir** [reply]
for now, yes. I might be able to make a plugin for native rounded corners but it could be a while until I do it.

Btw if you use https://github.com/tauri-apps/tauri-plugin-shadows you will get shadows and rounded corners for frameless window on Windows 11

**Reply by xuchaoqian** [reply]
Thank you so much.

There are some paint points to create custom titlebars: 
1. It's tedious to build rounded-corners frameless windows
2. Double-click data-tauri-drag-region to toggle "maximize" status of windows without decorations not working correctly, [see here](https://github.com/tauri-apps/tauri/issues/3220)
3. Lost native features on macOS when hover on traffic lights of inactive windows

**Reply by amrbashir** [reply]
1. hopefully will find a suitable solution
2. it should be working. will comment on the issue you opened.
3. should be possible to keep traffic lights and have frameless window, If you didn't find an issue for it, please open one.

**Reply by xuchaoqian** [reply]
> 3. should be possible to keep traffic lights and have frameless window, If you didn't find an issue for it, please open one.

[related issue](https://github.com/tauri-apps/tauri/issues/2663)

### Active frameless window
https://user-images.githubusercontent.com/1064052/149648097-9fb61960-fc9e-4077-be0c-58550bd83851.mov

### Inactive frameless window
https://user-images.githubusercontent.com/1064052/149648196-1ebecd09-fc48-4c83-be53-667900e49e6c.mov

**Reply by amrbashir** [reply]
it is not implemented in tauri yet but it is possible to keeep the native traffic lights

**Reply by xuchaoqian** [reply]
> it is not implemented in tauri yet but it is possible to keeep the native traffic lights

Good news!

Customizing titlebar is a very common use case for desktop apps. Hopefully tauri team will pay more attention to it.

**Comment by ngdangtu-vn** [comment]
Do we get border-radius for decoration-less windows yet?

**Reply by GlowFRA** [reply]
did you find a solution ?

**Reply by ngdangtu-vn** [reply]
Still the same me few months ago. Not yet.

**Reply by tekigg** [reply]
I used https://github.com/tauri-apps/window-shadows which is a workaround and I don't know if it has any customization but it adds round corners on windows 11 plus shadows and borders
